File: Array/0565.Array Nesting/Solution.java
class Solution {
    public int arrayNesting(int[] nums) {
        int n = nums.length, ans = 0;
        boolean[] used = new boolean[n];
        for (int i = 0; i < n; i++) {
            int cur = nums[i], cnt = 0;
            while (!used[cur]) {
                used[cur] = true;
                cur = nums[cur];
                cnt++;
            }
            ans = Math.max(ans, cnt);
        }
        return ans;
    }
}

File: Array/0565.Array Nesting/Solution.py
class Solution:
    def arrayNesting(self, nums: List[int]) -> int:
        ans, n = 0, len(nums)
        used = [False] * n
        for i in range(n):
            cur, cnt = nums[i], 0
            while not used[cur]:
                cnt += 1
                used[cur] = True
                cur = nums[cur]
            ans = max(ans, cnt)
        return ans